tests/lib/util/source-code.js

```diff
@@ -928,19 +988,488 @@ describe("SourceCode", () => {
+            eslint.on("CallExpression", assertCommentCount(0, 0));
+            eslint.on("BinaryExpression", assertCommentCount(0, 0));
+            eslint.on("Literal", assertCommentCount(0, 0));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return leading comments if the code only contains comments", () => {
+            const code = [
+                "//comment",
+                "/*another comment*/"
+            ].join("\n");
+
+            eslint.on("Program", assertCommentCount(2, 0));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return trailing comments if a block statement only contains comments", () => {
+            const code = [
+                "{",
+                "    //comment",
+                "    /*another comment*/",
+                "}"
+            ].join("\n");
+
+            eslint.on("Program", assertCommentCount(0, 0));
+            eslint.on("BlockStatement", assertCommentCount(0, 2));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return trailing comments if a class body only contains comments", () => {
+            const code = [
+                "class Foo {",
+                "    //comment",
+                "    /*another comment*/",
+                "}"
+            ].join("\n");
+
+            eslint.on("Program", assertCommentCount(0, 0));
+            eslint.on("ClassDeclaration", assertCommentCount(0, 0));
+            eslint.on("ClassBody", assertCommentCount(0, 2));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return trailing comments if an object only contains comments", () => {
+            const code = [
+                "({",
+                "    //comment",
+                "    /*another comment*/",
+                "})"
+            ].join("\n");
+
+            eslint.on("Program", assertCommentCount(0, 0));
+            eslint.on("ExpressionStatement", assertCommentCount(0, 0));
+            eslint.on("ObjectExpression", assertCommentCount(0, 2));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return trailing comments if an array only contains comments", () => {
+            const code = [
+                "[",
+                "    //comment",
+                "    /*another comment*/",
+                "]"
+            ].join("\n");
+
+            eslint.on("Program", assertCommentCount(0, 0));
+            eslint.on("ExpressionStatement", assertCommentCount(0, 0));
+            eslint.on("ArrayExpression", assertCommentCount(0, 2));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return trailing comments if a switch statement only contains comments", () => {
+            const code = [
+                "switch (foo) {",
+                "    //comment",
+                "    /*another comment*/",
+                "}"
```

Comment:
This all makes sense to me, and the new tests verify the behavior, let's go with it :+1: